metau-capstone/API/APIManager.m

```diff
@@ -0,0 +1,51 @@
+//
+//  APIManager.m
+//  metau-capstone
+//
+//  Created by Shirley Zhu on 7/18/22.
+//
+
+#import "APIManager.h"
+
+static NSString * const baseURLString = @"https://sheets.googleapis.com";
+
+@interface APIManager()
+
+@end
+
+@implementation APIManager
++ (instancetype)shared {
+    static APIManager *sharedManager = nil;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        sharedManager = [[self alloc] init];
+    });
+    return sharedManager;
+}
+
+- (instancetype)init {
+    self.baseURL = [NSURL URLWithString:baseURLString];
+    
+    NSString *path = [[NSBundle mainBundle] pathForResource: @"Info" ofType: @"plist"];
+    NSDictionary *dict = [NSDictionary dictionaryWithContentsOfFile: path];
+    self.APIkey = [dict objectForKey: @"sheets_api_key"];
+    NSLog(@"API: %@", self.APIkey);
+    
+    return self;
+}
+
+- (void) getSheetsData: (void(^)(NSError *error))completion {
```

Comment:
when design api interface to be invoked by other callsites, often worth to consider what is expected to return. In this case you likely need to handle the output of sheet api as well as error if any.
Couple ways to handle this:
1. have separate blocks handler; one for success case and one for error case
2. have a single handler that handle a response object which contains both the data & error; on success case the data part is non-nil and error would be nil; and vice versa for the the error case

Reply:
done! I updated the completion handler to contain both the data and error